### ISSUE TITLE
Exec format error

### ISSUE THREAD

**Issue by randoum** [tomaspinho/rtl8821ce#119]
Hi.

It is impossible to install 8821CE module on my laptop, I receive the message `modprobe: ERROR: could not insert '8821ce': Exec format error`

This is how I compiled it :
```
# git clone https://github.com/tomaspinho/rtl8821ce
Cloning into 'rtl8821ce'...

# cd rtl8821ce
# ./dkms-install.sh
About to run dkms install steps...

Creating symlink /var/lib/dkms/rtl8821ce/v5.5.2_34066.20190614/source ->
                 /usr/src/rtl8821ce-v5.5.2_34066.20190614

DKMS: add completed.

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
'make' -j4 KVER=4.19.0-5-amd64..............
cleaning build area...

DKMS: build completed.

8821ce.ko:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/4.19.0-5-amd64/updates/dkms/

depmod...

DKMS: install completed.
Finished running dkms install steps.
```

And when trying to activate it:
```
# modprobe 8821ce
modprobe: ERROR: could not insert '8821ce': Exec format error
```

Verification:

```
# dkms status
rtl8821ce, v5.5.2_34066.20190614, 4.19.0-5-amd64, x86_64: installed

# uname -a
Linux benj-p 4.19.0-5-amd64 #1 SMP Debian 4.19.37-5+deb10u2 (2019-08-08) x86_64 GNU/Linux
```

And my modules:
```
# lsmod | sort
ac                     16384  0
acpi_pad               24576  0
acpi_thermal_rel       16384  1 int3400_thermal
aesni_intel           200704  2
aes_x86_64             20480  1 aesni_intel
ahci                   40960  0
ansi_cprng             16384  0
autofs4                49152  2
battery                20480  0
binfmt_misc            20480  1
bluetooth             647168  41 btrtl,btintel,btbcm,bnep,btusb,rfcomm
bnep                   24576  2
btbcm                  16384  1 btusb
btintel                24576  1 btusb
btrfs                1384448  0
btrtl                  16384  1 btusb
btusb                  53248  0
button                 16384  0
cfg80211              761856  2 rtlwifi,mac80211
cmac                   16384  1
coretemp               16384  0
crc16                  16384  2 bluetooth,ext4
crc32c_generic         16384  0
crc32c_intel           24576  3
crc32_pclmul           16384  0
crct10dif_pclmul       16384  0
cryptd                 28672  3 crypto_simd,ghash_clmulni_intel,aesni_intel
crypto_simd            16384  1 aesni_intel
drbg                   28672  1
drm                   483328  5 drm_kms_helper,i915
drm_kms_helper        200704  1 i915
ecb                    16384  0
ecdh_generic           24576  2 bluetooth
efi_pstore             16384  0
efivarfs               16384  1
efivars                20480  1 efi_pstore
evdev                  28672  13
ext4                  733184  1
fat                    86016  1 vfat
fscrypto               32768  1 ext4
fuse                  122880  3
ghash_clmulni_intel    16384  0
glue_helper            16384  1 aesni_intel
hp_accel               28672  0
hp_wireless            16384  0
hp_wmi                 16384  0
i2c_algo_bit           16384  1 i915
i2c_i801               28672  0
i915                 1728512  16
input_polldev          16384  1 lis3lv02d
int3400_thermal        16384  0
int3403_thermal        16384  0
int340x_thermal_zone    16384  2 int3403_thermal,processor_thermal_device
intel_cstate           16384  0
intel_pch_thermal      16384  0
intel_powerclamp       16384  0
intel_rapl             24576  0
intel_rapl_perf        16384  0
intel_soc_dts_iosf     16384  1 processor_thermal_device
intel_uncore          135168  0
ip_tables              28672  0
irqbypass              16384  1 kvm
iTCO_vendor_support    16384  1 iTCO_wdt
iTCO_wdt               16384  0
jbd2                  122880  1 ext4
joydev                 24576  0
kvm                   724992  0
libahci                40960  1 ahci
libata                270336  2 libahci,ahci
libcrc32c              16384  1 btrfs
libphy                 77824  2 r8169
lis3lv02d              24576  1 hp_accel
lp                     20480  0
mac80211              815104  3 rtl_pci,rtl8188ee,rtlwifi
mbcache                16384  1 ext4
media                  45056  2 videodev,uvcvideo
mei                   118784  1 mei_me
mei_me                 45056  0
Module                  Size  Used by
nls_ascii              16384  1
nls_cp437              20480  1
nvme                   36864  3
nvme_core              81920  5 nvme
parport                57344  3 parport_pc,lp,ppdev
parport_pc             32768  0
pcc_cpufreq            16384  0
pcspkr                 16384  0
ppdev                  20480  0
processor_thermal_device    16384  0
psmouse               172032  0
r8169                  86016  0
raid6_pq              122880  1 btrfs
rfcomm                 86016  16
rfkill                 28672  6 hp_wmi,bluetooth,cfg80211
rng_core               16384  1 tpm
rtl8188ee             106496  0
rtl_pci                36864  1 rtl8188ee
rtlwifi                98304  2 rtl_pci,rtl8188ee
scsi_mod              245760  1 libata
serio_raw              16384  0
snd                    94208  18 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hwdep,snd_hda_intel,snd_hda_codec,snd_hda_codec_realtek,snd_timer,snd_compress,snd_soc_core,snd_pcm
snd_compress           24576  1 snd_soc_core
snd_hda_codec         151552  4 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_codec_realtek
snd_hda_codec_generic    86016  1 snd_hda_codec_realtek
snd_hda_codec_hdmi     57344  1
snd_hda_codec_realtek   122880  1
snd_hda_core           94208  7 snd_hda_codec_generic,snd_hda_codec_hdmi,snd_hda_intel,snd_hda_ext_core,snd_hda_codec,snd_hda_codec_realtek,snd_soc_skl
snd_hda_ext_core       28672  1 snd_soc_skl
snd_hda_intel          45056  4
snd_hwdep              16384  1 snd_hda_codec
snd_pcm               114688  7 snd_hda_codec_hdmi,snd_hda_intel,snd_hda_ext_core,snd_hda_codec,snd_soc_core,snd_soc_skl,snd_hda_core
snd_soc_acpi           16384  2 snd_soc_acpi_intel_match,snd_soc_skl
snd_soc_acpi_intel_match    24576  1 snd_soc_skl
snd_soc_core          253952  1 snd_soc_skl
snd_soc_skl           114688  0
snd_soc_skl_ipc        73728  1 snd_soc_skl
snd_soc_sst_dsp        36864  1 snd_soc_skl_ipc
snd_soc_sst_ipc        16384  1 snd_soc_skl_ipc
snd_timer              36864  1 snd_pcm
soundcore              16384  1 snd
sparse_keymap          16384  1 hp_wmi
thermal                20480  0
tpm                    65536  3 tpm_tis,tpm_crb,tpm_tis_core
tpm_crb                16384  0
tpm_tis                16384  0
tpm_tis_core           20480  1 tpm_tis
usb_common             16384  1 usbcore
usbcore               290816  4 xhci_hcd,uvcvideo,btusb,xhci_pci
uvcvideo              118784  0
vfat                   20480  1
video                  45056  1 i915
videobuf2_common       53248  2 videobuf2_v4l2,uvcvideo
videobuf2_memops       16384  1 videobuf2_vmalloc
videobuf2_v4l2         28672  1 uvcvideo
videobuf2_vmalloc      16384  1 uvcvideo
videodev              212992  3 videobuf2_v4l2,uvcvideo,videobuf2_common
wmi                    28672  2 hp_wmi,wmi_bmof
wmi_bmof               16384  0
x86_pkg_temp_thermal    16384  0
xhci_hcd              266240  1 xhci_pci
xhci_pci               16384  0
xor                    24576  1 btrfs
x_tables               45056  1 ip_tables
xxhash                 16384  2 zstd_compress,zstd_decompress
zstd_compress         172032  1 btrfs
zstd_decompress        81920  1 btrfs
```

If you have any idea on what may have caused this issue?
Thanks!!

**Comment by dallenwilson** [comment]
Double-check you've got the correct linux-headers package installed for that kernel.

Assuming you do, try forcing it:
`modprobe --force 8812ce`

Also, what shows up in dmesg after you try to load the module?
`dmesg | grep '8821ce'`

**Comment by randoum** [comment]
Hi @dallenwilson 

The header are uptodate 

```
# apt show linux-headers-$(uname -r)
Package: linux-headers-4.19.0-5-amd64
Version: 4.19.37-5+deb10u2
Status: install ok installed
(...)
```

Forcing gives the same output

```
#  modprobe --force 8821ce
modprobe: ERROR: could not insert '8821ce': Exec format error
```

And this is the dmesg

```
#  dmesg | grep '8821ce'
[    4.846843] 8821ce: loading out-of-tree module taints kernel.
[    4.847610] 8821ce: module verification failed: signature and/or required key missing - tainting kernel
[10182.592520] 8821ce: module_layout: kernel tainted.
```

I also tried to `dkms remove` and `dkms install` the module with no luck.

**Comment by 9Lukas5** [comment]
@randoum  Maybe i'm totally wrong, just a guess that popped in my mind: Do you have a secure boot enabled system?

**Comment by randoum** [comment]
Yes, disabled in the BIOS, first thing I checked :)

```
# dmesg | grep -i secure
[    0.000000] secureboot: Secure boot could not be determined (mode 0)
[    1.270539] Loaded X.509 cert 'Debian Secure Boot CA: 6ccece7e4c6c0d1f6149f3dd27dfcc5cbb419ea1'
[    1.270550] Loaded X.509 cert 'Debian Secure Boot Signer: 00a7468def'
[    1.272300] Loaded UEFI:db cert 'Hewlett-Packard Company: HP UEFI Secure Boot 2013 DB key: 1d7cf2c2b92673f69c8ee1ec7063967ab9b62bec' linked to secondary sys keyring
```

**Comment by 9Lukas5** [comment]
Just for clarification for me: do you just have the problem of modprobing doesn't work, but the wifi works after `dkms install` and reboot, or the wifi doesn't work after reboot?

**Comment by randoum** [comment]
Fixed by updating kernel to 4.19.0-6-amd64 (available since today)
Compatibility issue ? Anyhow it's working now.

Thanks to all who send time to help me!! :heart:
